Apply PE Datastructure and include Matrix-like Calculation

heyyeahcrow opened the pull request:
In the function arrhenius_rate of newArrheniusEq.jl, I wanted to set the k_matrix as StaticMatrix as the energies in PhaseEnergies. However, the diagonal values in k_matrix are not adaptable without being assigned as Array. I'm wondering if there's any better solutions

heyyeahcrow committed:
Apply PE Datastructure and include

heyyeahcrow committed:
Delete 3phases.ipynb

rkurchin commented on Simulation.jl:
```diff
@@ -1,4 +1,10 @@
-function simulate_deposition(compositions_T, fcoeff, Ea_values, T, num_steps, dt)
+function meshgrid(x, y)
```
I don't have any issue with keeping this for convenience, but FYI, see this discussion about this type of functionality: https://discourse.julialang.org/t/meshgrid-function-in-julia/48679/15

rkurchin commented:
You could just change from `SMatrix` to `MMatrix`, which is still statically _sized_, but is mutable rather than static in its elements.

This generally looks good to me! Maybe make that one little change and merge it. I think the next thing to focus on is converting to a package structure with some tests before we start adding other "fancy" functionality. We can talk about that in our meeting today. 😃

rkurchin commented:
I would go ahead and make that `MMatrix` change and then merge this. I have another branch where I've started to set up the package directory structure and I can rebase it once this is merged so it has the latest versions of the files...

heyyeahcrow committed:
MMatrix for K

MMatrix seems not to be enough to adapt pe.K, so I also assign the structure to be mutable.
It works and is ready to be merged
